lib/go/thrift/type.go

```diff
@@ -38,9 +38,7 @@ const (
 	MAP    = 13
 	SET    = 14
 	LIST   = 15
-	UTF8   = 16
-	UTF16  = 17
-	//BINARY = 18   wrong and unusued
+	UUID   = 16
```

Comment:
for Java it's 17, FYI

Reply:
It's [16 in the spec](https://github.com/apache/thrift/blob/0223d6346675c5089c2a2de5fb6f3858e599c9a9/doc/specs/thrift-binary-protocol.md?plain=1#L204). does java pass cross test?

Reply:
https://github.com/apache/thrift/blob/master/lib/java/src/main/java/org/apache/thrift/protocol/TType.java#L37-L38

Reply:
@fishy Crosstests until recently were only running Java/Kotlin so obviously that was successful running crosstests against themselves even with incorrect spec. Swift was recently added but was implemented to match Java so also has UUID=17 so all those tests were passing

Reply:
UUID=17 has already been included in a stable release. We should either correct that with a breaking change, or update the spec and the Go implementation to be consistent.

Neither option is attractive.

Reply:
>UUID=17 has already been included in a stable release

is that true? I thought all UUID related changes are merged **after** 0.18.0.

Reply:
> is that true? I thought all UUID related changes are merged after 0.18.0.

I only did a cursory check. `TType.java` was last modified on 9 June with `UUID=17` and 0.17.0 was released on 18 September.

Reply:
NVM, the commit date is not the merge date. We didn't include it in a stable release. Sorry :)

Reply:
looking at https://github.com/apache/thrift/commits/v0.17.0 I _think_ the only thing uuid related included in the stable release was the spec, so we definitely should fix java/kotlin/swift to match the spec instead :)

Reply:
tracked in https://issues.apache.org/jira/browse/THRIFT-5653